micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/UrlUtils.java

```diff
@@ -0,0 +1,41 @@
+package com.ofg.infrastructure.web.resttemplate.fluent;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class UrlUtils {
+
+    private static final String FIRST_PARAMETER_CONCATENATOR = "?";
+    private static final String PARAMETERS_ASSIGNMENT = "=";
+    private static final String SECOND_AND_LATER_CONCATENATOR = "&";
+
+    public static URI addQueryParametersToUri(URI uri, Map<String, Object> params) throws URISyntaxException {
+        StringBuilder urlParametersBuilder = new StringBuilder();
+        if (uri == null) {
+            throw new IllegalArgumentException("Define URL before URL parameters");
+        }
+        urlParametersBuilder.append(uri.toString());
+        String concatenator = FIRST_PARAMETER_CONCATENATOR;
+        for (String paramName : params.keySet()) {
```

Comment:
Definitely you are ;). It looks that because it is a code in Java, not Groovy :). Piotrek did that loop in pain to adjust to the new policy and he rejected all Groovy or Java 8 goodness.